presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -211,7 +213,11 @@ else if (type.equals(DATE)) {
                     prefilledValue = datePartitionKey(columnValue, name);
                 }
                 else if (type.equals(TIMESTAMP) || type.equals(TIMESTAMP_WITH_TIME_ZONE)) {
-                    prefilledValue = timestampPartitionKey(columnValue, hiveStorageTimeZone, name, type.equals(TIMESTAMP_WITH_TIME_ZONE));
+                    DateTimeZone zone = hiveStorageTimeZone;
+                    if (isFileModifiedTimeColumnHandle(column)) {
+                        zone = UTC;
+                    }
```

Comment:
I originally said UTC to answer "but which zone?" (`timestamp with time zone` requires _some zone_, but we have just point in time), but the zone doesn't really matter.
We could keep it simpler and leave this code piece intact.